routers/web/repo/setting.go

```diff
@@ -911,6 +911,23 @@ func CollaborationPost(ctx *context.Context) {
 		return
 	}
 
+	// find the owner team of the organization the repo belongs too and
+	// check if the user we're trying to add is an owner.
+	if ctx.Repo.Repository.Owner.IsOrganization() {
+		teams, err := organization.GetRepoTeams(ctx, ctx.Repo.Repository)
+		if err != nil {
+			ctx.ServerError("GetRepoTeams", err)
+			return
+		}
+
+		for _, team := range teams {
+			if team.IsOwnerTeam() && team.IsMember(u.ID) {
+				ctx.Redirect(setting.AppSubURL + ctx.Req.URL.EscapedPath())
+				return
+			}
+		}
```

Comment:
This should use the `IsOrganizationOwner` function:
https://github.com/go-gitea/gitea/blob/e91229eefbd09fecaebd0e058c3bfe7612a0bc61/models/organization/org_user.go#L44-L55

Reply:
👍